hudi-integ-test/src/test/java/org/apache/hudi/integ/HoodieTestHiveBase.java

```
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+
+package org.apache.hudi.integ;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.util.collection.Pair;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Base class to run cmd and generate data in hive.
+ */
+public class HoodieTestHiveBase extends ITTestBase {
+
+  protected enum PartitionType {
+    SINGLE_KEY_PARTITIONED, MULTI_KEYS_PARTITIONED, NON_PARTITIONED,
+  }
+
+  /**
+   * A basic integration test that runs HoodieJavaApp to create a sample Hoodie data-set and performs upserts on it.
+   * Hive integration and upsert functionality is checked by running a count query in hive console. TODO: Add
+   * spark-shell test-case
+   */
+  public void generateDataByHoodieJavaApp(String hiveTableName, String tableType, PartitionType partitionType,
+      String commitType, String hoodieTableName) throws Exception {
+
+    String hdfsPath = getHdfsPath(hiveTableName);
+    String hdfsUrl = "hdfs://namenode" + hdfsPath;
+
+    Pair<String, String> stdOutErr;
+    if ("overwrite".equals(commitType)) {
+      // Drop Table if it exists
+      try {
+        dropHiveTables(hiveTableName, tableType);
+      } catch (AssertionError ex) {
+        // In travis, sometimes, the hivemetastore is not ready even though we wait for the port to be up
+        // Workaround to sleep for 5 secs and retry
+        // Set sleep time by hoodie.hiveserver.time.wait
+        Thread.sleep(getTimeWait());
+        dropHiveTables(hiveTableName, tableType);
+      }
+
+      // Ensure table does not exist
+      stdOutErr = executeHiveCommand("show tables like '" + hiveTableName + "'");
+      assertTrue(stdOutErr.getLeft().isEmpty(), "Dropped table " + hiveTableName + " exists!");
```

I would not suggest using assertation in normal methods. We can throw an exception directly. WDYT?